src/lua.rs

```diff
@@ -625,6 +626,69 @@ impl Lua {
         }
     }
 
+    /// Set a function for Lua to call on conditions configured by `options`. This function will
+    /// always succeed, but the hook may raise an error to be reported back to the caller when
+    /// running Lua code.
+    ///
+    /// # Example
+    ///
+    /// Shows each line number of code being executed by the Lua interpreter.
+    ///
+    /// ```
+    /// # extern crate rlua;
+    /// # use rlua::{Lua, HookOptions, Debug};
+    /// # fn main() {
+    /// let code = r#"local x = 2 + 3
+    /// local y = x * 63
+    /// local z = string.len(x..", "..y)
+    /// "#;
+    ///
+    /// let lua = Lua::new();
+    /// lua.set_hook(HookOptions {
+    ///     lines: true, ..Default::default()
+    /// }, |debug: &Debug| {
+    ///     println!("line {}", debug.curr_line);
+    ///     Ok(())
+    /// });
+    /// let _ = lua.exec::<_, ()>(code, None);
+    /// # }
+    /// ```
+    pub fn set_hook<F>(
+        &self,
+        options: HookOptions,
+        callback: F)
+    where
+        F: 'static + Send + Fn(&Debug) -> Result<()>
+    {
+        self.set_mut_hook(options, callback)
+    }
+
+    /// Set a function for Lua the same way as [`set_hook`], but allows the use of a mutable
+    /// function. The behavior is the same otherwise.
+    ///
+    /// [`set_hook`]: #method.set_hook
+    pub fn set_mut_hook<F>(
```

Comment:
I think this method is redundant, since every closure that implements `Fn` also implements `FnMut`, so just having one `set_hook` that takes an `FnMut` should be enough.